assets/src/edit-story/components/panels/link.js

```diff
@@ -82,29 +84,40 @@ function LinkPanel({ selectedElements, onSetProperties }) {
   );
   const handleSubmit = useCallback(
     (evt) => {
-      onSetProperties({ link: state?.url ? state : null });
+      onSetProperties({ link: state?.url ? { ...state } : null });
       if (evt) {
         evt.preventDefault();
       }
     },
     [state, onSetProperties]
   );
 
+  const {
+    actions: { getLinkMetadata },
+  } = useAPI();
+
   const canLink = selectedElements.length === 1 && !isFill;
-  const populateMetadata = useCallback(
-    debounce(300, async (/** url */) => {
-      // TODO(wassgha): Implement getting the page metadata
-    })
-  );
+
+  const [populateMetadata] = useDebouncedCallback((url) => {
+    setFetchingMetadata(true);
+    getLinkMetadata(url).then(({ title, image }) => {
+      setState((originalState) => ({
```

Comment:
Should this do anything differently when the link is actually incorrect and data was not received?